public/static/js/spawned_hosts.js

```diff
@@ -439,8 +439,20 @@ mciModule.controller('SpawnedHostsCtrl', ['$scope', '$window', '$timeout', '$q',
       $scope.selectedDistro = spawnableDistro;
       $scope.spawnInfo.distroId = spawnableDistro.name;
 
-      if ($scope.selectedDistro.regions.length > 0) {
-        $scope.selectedRegion = $scope.selectedDistro.regions[0];
+      // if multiple regions, preference the user region
+      if ($scope.selectedDistro.regions.length > 1) {
+        if ($scope.region !== "") {
+          for (let i=0; i < $scope.selectedDistro.regions.length; i++) {
+            // valid region
+            if ($scope.region === $scope.selectedDistro.regions[i]) {
+              $scope.selectedRegion = $scope.region;
+            }
+          }
+        }
+        // if preferred region not configured for this distro, default to the first region in this list
+        if ($scope.selectedRegion === undefined) {
+          $scope.selectedRegion = $scope.selectedDistro.regions[0];
```

Comment:
If the distro does not have any regions, does this default to the global ones?

Reply:
This block only gets called if the distro has multiple regions defined. Otherwise for EC2 hosts we default to "us-east-1" (Evergreen's default) but we don't show this to the user. Should we? I think it's okay to hide this since it's not configurable, and the availability zone will still be shown on the host page for the user.